pkg/gui/controllers/helpers/cherry_pick_helper.go

```diff
@@ -90,6 +90,7 @@ func (self *CherryPickHelper) Paste() error {
 				if err := self.c.Git().Rebase.CherryPickCommitsDuringRebase(self.getData().CherryPickedCommits); err != nil {
 					return err
 				}
+				_ = self.Reset()
```

Comment:
The lack of error handling makes me slightly nervous. If a function can return an error, it should be handled; if the function never returns an error, or it's not important to handle, then it shouldn't have a return type of "error".

One way to solve this could be to remove the Reset from here, turn the two `return`s below into `err = ...` (the second would have to be in an `else` block then), and at the very end of the function do
```
if err == nil {
  err = Reset()
}
return err
```

Reply:
OK, that suggestion doesn't work, because of the `WithWaitingStatus` in the else branch, which is asynchronous.

But we can still do the Reset after carrying out the operation, and only if it didn't return an error, we just have to do it in both places. I pushed a fixup to do that (ea903598cb06e); and I also included an additional check to only clear the commits if we didn't get conflicts, to address the concern I brought up above. I feel this is a good middle ground.

Let me know what you think!

Reply:
Thank you @stefanhaller for your feedback and your fix.

I tested the proposed solution in the sandbox and it makes a lot of sense to me too.
Currently there is a slight inconsistency in the behavior (after having solved merge conflicts user still needs to manually press Esc to clear the copied commits), but I think it's an acceptable trade-off.

Reply:
I agree that the trade-off is acceptable. Just waiting for @jesseduffield's ok before I merge.